hera_cal/io.py

```diff
@@ -312,6 +323,14 @@ def __init__(self, input_data, filetype='uvh5', **read_kwargs):
         else:
             raise NotImplementedError('Filetype ' + self.filetype + ' has not been implemented.')
 
+        # save longest and shortest integration times in the file for later use in up/downsampling
+        # if available, these will be used instead of the ones in self.integration_time during partial I/O
+        self.longest_integration = None
+        self.longest_integration = None
+        if self.integration_time is not None:
+            self.longest_integration = np.max(self.integration_time)
```

Comment:
the only way this may break is if the integration_time is updated downstream to track coherent averaging. Currently we do this by adding to `nsamples` but I vaguely recall a conversation years ago about possibly changing integration_time?